### ISSUE TITLE
Cannot reach test-net IP address space

### ISSUE THREAD

**Issue by Marc05** [safing/portmaster#1110]
Pings fail even with global rules allowing the traffic:
```
Pinging 192.0.2.2 with 32 bytes of data:
Reply from 127.0.0.1: Destination port unreachable.
Reply from 127.0.0.1: Destination port unreachable.
Reply from 127.0.0.1: Destination port unreachable.
Reply from 127.0.0.1: Destination port unreachable.

Ping statistics for 192.0.2.2:
    Packets: Sent = 4, Received = 4, Lost = 0 (0% loss),
```

![image](https://user-images.githubusercontent.com/1699133/220733623-134885b3-9e2c-4413-b840-ed3c1d536c03.png)

**Comment by Raphty** [comment]
sorry for the "dum" question, but Pormaster says invalid IP are you sure that you are pinging the correct IP?

also @dhaavi  I know that we had some discussion about ping - I think SPN does not support it but normal Portmaster should not have any issues?

**Comment by Marc05** [comment]
It's being done on purpose - see:
https://github.com/safing/portmaster/blob/02779f8bfd91c11bde6c60e7e05a81f6bb24c718/network/netutils/ip.go#L51

I'm not sure what the reason behind it is because these are legitimate addresses.

**Comment by TFWol** [comment]
That's not a good range to use.
They are adhering to RFC5757 where it states "[These blocks are not for local use ](https://datatracker.ietf.org/doc/html/rfc5737#section-4)"

**Comment by Marc05** [comment]
The title of the issue report mentions "test-net", and the code I linked references the relevant RFC - that's a long way of saying I'm aware of what it is; excuse the annoyed tone. 

I keep finding things from small annoyances to workflow-breaking issues like this, in addition to the development not being as active as I expected for it being a 1.0 release. I'll wait for some feedback from the devs, but I suppose this may not be the right product for me.

**Comment by dhaavi** [comment]
Our behavior is in line with RFC5757, which states:

      3.  Documentation Address Blocks

         The blocks 192.0.2.0/24 (TEST-NET-1), 198.51.100.0/24 (TEST-NET-2),
         and 203.0.113.0/24 (TEST-NET-3) are provided for use in
         documentation.

      4.  Operational Implications

         Addresses within the TEST-NET-1, TEST-NET-2, and TEST-NET-3 blocks
         SHOULD NOT appear on the public Internet and are used without any
         coordination with IANA or an Internet registry [RFC2050].  Network
         operators SHOULD add these address blocks to the list of non-
         routeable address spaces, and if packet filters are deployed, then
         this address block SHOULD be added to packet filters.

         These blocks are not for local use, and the filters may be used in
         both local and public contexts.

Can you please elaborate on your use case?

**Comment by Marc05** [comment]
These are commonly used in labs (in place of real public IP address space), which is my use-case. I understand the RFC states they _shouldn't_ be used in local/public contexts, and that network operators should be block these in packet filters. My argument is that a lab context is a valid practical use case, and that the decision should be left to the "network operator", not forced by Portmaster. I propose an advanced option (default disabled) to allow all of the current networks marked as invalid by portmaster.

**Comment by Marc05** [comment]
Fixed in 1.4.7 - now I can allow the traffic with rules:
https://github.com/safing/portmaster/commit/81c801237db0bc39a6e273e1b5649b136e146878 Move blocking of invalid IPs behind rules
Thank you!
